crates/uv-toolchain/src/managed.rs

```diff
@@ -205,14 +217,25 @@ impl InstalledToolchain {
             .to_str()
             .ok_or(Error::NameError("not a valid string".to_string()))?
             .to_string();
-        let python_version = PythonVersion::from_str(key.split('-').nth(1).ok_or(
-            Error::NameError("not enough `-`-separated values".to_string()),
+        let parts: Vec<&str> = key.split('-').collect();
+        let implementation = ImplementationName::from_str(parts.first().ok_or(
+            Error::NameParseError(key.clone(), "not enough `-`-separated values".to_string()),
         )?)
-        .map_err(|err| Error::NameError(format!("invalid Python version: {err}")))?;
+        .map_err(|err| {
+            Error::NameParseError(key.clone(), format!("invalid Python implementation: {err}"))
+        })?;
+        let python_version = PythonVersion::from_str(parts.get(1).ok_or(Error::NameParseError(
```

Comment:
Do you expect a fixed number of parts? You could do `let [implementation, python_version] = parts.as_slice() else { ... }` if so.

Reply:
Ah yeah totally that'd be nice.